# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/warp/bug1646302.js
# Commit: 704d04e6bc76
# Full Hash: 704d04e6bc76e06107a2a8856cef2363f70f73fb
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2020-06-24 16:24:33
# Description:
#   Bug 1646302 - Only create a template object if prototype is an object in getTemplateObjectForScripted. r=evilpie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D80595
# ==============================================================================

diff -r e29a263455f8 -r 704d04e6bc76 js/src/jit-test/tests/warp/bug1646302.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/warp/bug1646302.js	Tue Jun 23 10:11:10 2020 +0000
@@ -0,0 +1,9 @@
+function f(x) {
+  function fnc() {}
+  fnc.prototype = 3;
+  new fnc;
+  if (x < 50) {
+    new new.target(x + 1);
+  }
+}
+new f(0);